docs/migration_guide.rst

```diff
@@ -1,19 +1,68 @@
 0.5.0 Migration guide
 =====================
 
-0.5.0 updates project to support cairo 0.10.0
+``cairo-lang`` 0.10.0 brings a lot of new exciting changes, like:
 
-New transaction version
+- new cairo syntax,
+- new transaction version (1),
+- new ``__validate__`` endpoint in accounts.
+
+``starknet.py`` 0.5.0 has an experimental support for new features and tries to minimize number of breaking changes for
+users who want to use the old transaction version (0). Please note that support for this transaction version will be
+removed in the future.
+
```

Comment:
Add a note that informs users that they can use the previous version if they don't need any new features.